spyder/plugins/ipythonconsole/comms/kernelcomm.py

```diff
@@ -108,40 +123,45 @@ def _get_call_return_value(self, call_dict, call_data, comm_id):
         """
         settings = call_dict['settings']
         interrupt = 'interrupt' in settings and settings['interrupt']
-        if interrupt:
-            with self.comm_channel_manager(comm_id):
-                return super(KernelComm, self)._get_call_return_value(
-                    call_dict, call_data, comm_id)
-        else:
+
+        with self.comm_channel_manager(comm_id, queue_message=not interrupt):
             return super(KernelComm, self)._get_call_return_value(
                 call_dict, call_data, comm_id)
 
     def _wait_reply(self, call_id, call_name, timeout):
         """Wait for the other side reply."""
-        if call_id in self._reply_inbox:
+
+        def got_reply():
+            return call_id in self._reply_inbox
+
+        timeout_msg = "Timeout while waiting for {}".format(
+                            self._reply_waitlist)
+        self._wait(got_reply, self._sig_got_reply, timeout_msg, timeout)
+
+    def _wait(self, condition, signal, timeout_msg, timeout):
+        """Wait for a condition"""
```

Comment:
Please expand this docstring to describe what `condition`, `signal`, `timeout_msg` and `timeout` are.